### DISCUSSION TITLE
Not find symbol PasswordPolicy.HASH_ALGORITHM_DEFAULT

### DISCUSSION THREAD

**Discussion by ajaiswal595** [keycloak/keycloak#33294] [Adapter deprecation]
We were using this in 24.0.4
import org.keycloak.models.PasswordPolicy;

private static final String DEFAULT_HASH_ALGORITH = PasswordPolicy.HASH_ALGORITHM_DEFAULT;

But now we get this error when we upgrade to 25.0.6 version

not find symbol
    private static final String DEFAULT_HASH_ALGORITH = PasswordPolicy.HASH_ALGORITHM_DEFAULT;
                                                                      ^
  symbol:   variable HASH_ALGORITHM_DEFAULT
  location: class PasswordPolicy
  
  
  What is the replacement of this ?

**Comment by codespearhead** [comment]
It was removed from `server-spi/src/main/java/org/keycloak/models/PasswordPolicy.java` by this PR [1].

What you want is probably  `org.keycloak.models.PasswordPolicy.getHashAlgorithm()` [2].

[1] https://github.com/keycloak/keycloak/pull/28128
[2] https://github.com/keycloak/keycloak/blob/b26febc32a5728e4cf4ff296b951f426a945bf51/server-spi/src/main/java/org/keycloak/models/PasswordPolicy.java#L78
